Failed to pass authentication with a presigned url

test_cors_presigned_get_object, test_cors_presigned_get_object_tenant

presigned url generated via boto3:

```
url = client.generate_presigned_url(
        ClientMethod="get_object", Params=params, ExpiresIn=100000, HttpMethod="GET"
)
```

url is equal to:
```
https://s3.neofs.devenv:8080/yournamehere-06xz3xodr0ok0evu-1/foo?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=2rXrksx5Mpie1mKMyFrx9ZBuPcS6hAjMd4MTAhA3xacS0CM6oSUDvnJzyuBPyxSKzrT88nKALN2c1A6vZDtjoapj9%2F20231108%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231108T021749Z&X-Amz-Expires=100000&X-Amz-SignedHeaders=host&X-Amz-Signature=d7d8a7ea80e7bf4303f9686d2d0e2288f9ff76f2a6244bf3b749c9f2a78749cf
```

Upon request got following error:
```
```

After https://github.com/nspcc-dev/neofs-s3-gw/pull/964 we have two another red tests
```
FAILED s3tests_boto3/functional/test_s3.py::test_object_raw_get_x_amz_expires_not_expired - assert 400 == 403
FAILED s3tests_boto3/functional/test_s3.py::test_object_raw_get_x_amz_expires_not_expired_tenant - assert 400 == 403
```
According to history [here](https://github.com/nspcc-dev/s3-tests/blob/7fff720faa477bcc87091a1905bfedd1ab6c7878/s3tests_boto3/functional/test_s3.py#L4292) status was changed from 400 to 403 for skipping. Do we want to revert to 400?

--- a/api/handler/cors.go
+++ b/api/handler/cors.go
@@ -136,6 +136,10 @@ func (h *handler) AppendCORSHeaders(w http.ResponseWriter, r *http.Request) {
 }
 
 func (h *handler) Preflight(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodOptions {
+		return
+	}
+
 	reqInfo := api.GetReqInfo(r.Context())
 	bktInfo, err := h.obj.GetBucketInfo(r.Context(), reqInfo.BucketName)
 	if err != nil {
@@ -146,6 +150,7 @@ func (h *handler) Preflight(w http.ResponseWriter, r *http.Request) {
 	origin := r.Header.Get(api.Origin)
 	if origin == "" {
 		h.logAndSendError(w, "origin request header needed", reqInfo, s3errors.GetAPIError(s3errors.ErrBadRequest))
+		return
 	}
 
 	method := r.Header.Get(api.AccessControlRequestMethod)

--- a/api/router.go
+++ b/api/router.go
@@ -150,6 +150,21 @@ func appendCORS(handler Handler) mux.MiddlewareFunc {
 	}
 }
 
+func appendPreflight(handler Handler) mux.MiddlewareFunc {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			handler.Preflight(w, r)
+			if www, ok := w.(*logResponseWriter); ok {
+				if www.statusCode != 0 {
+					return
+				}
+			}
+
+			h.ServeHTTP(w, r)
+		})
+	}
+}
+
 func logErrorResponse(l *zap.Logger) mux.MiddlewareFunc {
 	return func(h http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
@@ -199,6 +214,8 @@ func Attach(r *mux.Router, domains []string, m MaxClients, h Handler, center aut
 
 		// -- logging error requests
 		logErrorResponse(log),
+
+		appendPreflight(h),
 	)
 
 	// Attach user authentication for all S3 routes.

